### PR TITLE
Add tools/format.sh which formats all files automatically.

### PR DESCRIPTION
**Pull request by g-easy**
Uses clang-format, cmake-format, and buildifier.

### CODE REVIEW

**Comment by g-easy** [comment]
Ryan, can you try this script on your PR #13?

Especially cmake-format and buildifier. :)

**Comment by g-easy** [comment]
@maxgolov what do you think? This kind of duplicates `git-cl.sh` - how should we resolve this?

In particular, I'd like something more automated than `git cl format specify_every_changed_file.cc`

File: tools/format.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+FIND="find . -name .git -prune -o -name _deps -prune -o -name .build -prune -o"
+
+# Correct common miscapitalizations.
+sed -i 's/Open[t]elemetry/OpenTelemetry/g' $($FIND -type f -print)
+# No CRLF line endings.
+sed -i 's/\r$//' $($FIND -type f -print)
+# No trailing spaces.
+sed -i 's/ \+$//' $($FIND -type f -print)
+
+CMD=clang-format-8
```
Comment (rnburn) [inline tools/format.sh]: Do we need to require clang-format-8? Could we use just clang-format here?

Comment (g-easy) [reply tools/format.sh]: That's what I started with, but on Debian stable clang-format is v5.0.0, which is too old to understand `AfterExternBlock` in our `.clang-format`

**Comment by rnburn** [comment]
I got this error when I ran on os x
```
$ ./tools/format.sh
sed: 1: "./tools/setup-buildtool ...": invalid command code .
```

File: tools/format.sh (hunk 1 above)
Comment (rnburn) [reply tools/format.sh]: Can we rely on the CI to catch an old clang-format? Or do a version check and error out if it's too low?

I have clang-format 8, but os x doesn't have a clang-format-8 alias.

**Comment by g-easy** [comment]
Looks like BSD vs GNU sed.

**Comment by g-easy** [comment]
Please try again. :)

**Comment by rnburn** [comment]
Worked!

File: tools/format.sh (hunk 1 above)
Comment (g-easy) [reply tools/format.sh]: Resolved by making the script more complicated. :^)

**Comment by g-easy** [comment]
Thank you for your patience, and testing this on OSX which I didn't have access to.

I think I need one more approval to merge.

**Pull request merged**